docker/enterprise/Dockerfile-jre11

```diff
@@ -0,0 +1,191 @@
+
+RUN \
+  curl \
+    --silent \
+    --location \
+    --retry 3 \
+    --output "/tmp/graylog-${GRAYLOG_VERSION}.tgz" \
+    "https://packages.graylog2.org/releases/graylog/graylog-${GRAYLOG_VERSION}.tgz"
+
+RUN \
+  curl \
+    --silent \
+    --location \
+    --retry 3 \
+    --output "/tmp/graylog-${GRAYLOG_VERSION}.tgz.sha256.txt" \
+    "https://packages.graylog2.org/releases/graylog/graylog-${GRAYLOG_VERSION}.tgz.sha256.txt"
+
+# Verify server and extract the Server
+RUN \
+  sha256sum --check "graylog-${GRAYLOG_VERSION}.tgz.sha256.txt"
+
+RUN \
+  mkdir /opt/graylog && \
+  tar --extract --gzip --file "/tmp/graylog-${GRAYLOG_VERSION}.tgz" --strip-components=1 --directory /opt/graylog
+
+
+# Download Enterprise and Integrations Plugins and Integrations Enterprise Plugins.
+RUN \
+  curl \
+    --silent \
+    --location \
+    --retry 3 \
+    --output "/tmp/graylog-enterprise-integrations-plugins-${GRAYLOG_VERSION}.tgz" \
+    "https://downloads.graylog.org/releases/graylog-enterprise-integrations/graylog-enterprise-integrations-plugins-${GRAYLOG_VERSION}.tgz"
+
+RUN tar --extract --gzip --file "/tmp/graylog-enterprise-integrations-plugins-${GRAYLOG_VERSION}.tgz" --strip-components=1 --directory /opt/graylog
+
+RUN \
+  curl \
+    --silent \
+    --location \
+    --retry 3 \
+    --output "/tmp/graylog-integrations-plugins-${GRAYLOG_VERSION}.tgz" \
+    "https://downloads.graylog.org/releases/graylog-integrations/graylog-integrations-plugins-${GRAYLOG_VERSION}.tgz"
+
+RUN tar --extract --gzip --file "/tmp/graylog-integrations-plugins-${GRAYLOG_VERSION}.tgz" --strip-components=1 --directory /opt/graylog
+
+
+RUN \
+  curl \
+    --silent \
+    --location \
+    --retry 3 \
+    --output "/tmp/graylog-enterprise-plugins-${GRAYLOG_VERSION}.tgz" \
+    "https://downloads.graylog.org/releases/graylog-enterprise/graylog-enterprise-plugins-${GRAYLOG_VERSION}.tgz"
+
+RUN tar --extract --gzip --file "graylog-enterprise-plugins-${GRAYLOG_VERSION}.tgz" --strip-components=1 --directory /opt/graylog
+
+
+RUN \
+  install \
+    --directory \
+    --mode=0750 \
+    /opt/graylog/data \
+    /opt/graylog/data/journal \
+    /opt/graylog/data/log \
+    /opt/graylog/data/config \
+    /opt/graylog/data/plugin \
+    /opt/graylog/data/data \
+    /opt/graylog/data/scripts
+
+# -------------------------------------------------------------------------------------------------
+#
+# final layer
+# use the smallest debian with headless openjdk and copying files from download layers
+FROM openjdk:11-jre-slim-buster
+
+ARG VCS_REF
+ARG GRAYLOG_VERSION
+ARG BUILD_DATE
+ARG GRAYLOG_HOME=/usr/share/graylog
+ARG GRAYLOG_USER=graylog
+ARG GRAYLOG_UID=1100
+ARG GRAYLOG_GROUP=graylog
+ARG GRAYLOG_GID=1100
+
+COPY --from=graylog-downloader /opt/graylog ${GRAYLOG_HOME}
+COPY config ${GRAYLOG_HOME}/data/config
+
+WORKDIR ${GRAYLOG_HOME}
+
+# hadolint ignore=DL3027,DL3008
+RUN \
+  echo "export JAVA_HOME=/usr/local/openjdk-11"     > /etc/profile.d/graylog.sh && \
```

Comment:
Wouldn't it possible to avoid duplicating the entire Dockerfile(s) by changing this based on a docker `--build-arg`?
 
E.g. something like `ARG FLAVOR=jre11`?

Reply:
Wow, I can't believe I didn't think to do that! Okay, yeah. That's not the only thing I changed, though. The java arguments are different. I can make that a parameter, too, though.

Reply:
The java flags are a little tricky. If we want people to be able to experiment with different GC settings, we can't just hardcode some of these. But if we pass it as an ARG, it's built into the image.

Reply:
Alright, I merged the Dockerfiles. Thanks for pointing that out.